test/spmd/test_tracing.py

```diff
@@ -0,0 +1,142 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates
+import torch
+
+from torch.distributed._spmd import CommTensor
+from torch.distributed.distributed_c10d import (
+    get_global_rank,
+    get_world_size,
+)
+from torch.fx.experimental.proxy_tensor import make_fx
+from torch.testing._internal.common_utils import run_tests
+from spmd.testing.common_utils import (  # type: ignore
+    DistTensorTestBase,
+    with_comms,
+)
+from spmd.tensor import DeviceMesh
+
+from typing import List
+
+
+class TraceDeviceMeshTest(DistTensorTestBase):
+    @property
+    def world_size(self) -> int:
```

Comment:
I think our CI could only get 4 GPUs (for budget reason), so world_size=8 might not work well with the GPU CI, it could only test CPU CI

Reply:
I am reducing this PR to use world size 4. Will add another test for cpu-only 8 process coverage.

Reply:
If 8 GPU is needed, we can have another conversation with Dev Infra team to ask for it. @mrshenli do we really want machines with GPUs more than 4? We can add a tag to enable tests in CI for scenarios when we need CI for cases when GPU>4. I remember PyTorch CI does have something similar.

Reply:
IIUC, this is not urgent at this point, but will help us to improve test coverage. Do we know how long it might take to get a 8GPU machine? If it is prompt, we can wait till we have real 3D device mesh use cases? cc @wanchaol @aazzolini

Reply:
As long as we can justify and get the budget. It should be very quick. Previously. Ke added new machines in https://github.com/pytorch/test-infra/pull/602. And to change our CI script(within Tau repo) should also straightforward. Adding to PyTorch CI can be another topics which I believe we already have a thread for that.

Reply:
Just a note on the current divergence of GPU number used in tests:
mesh tests now are using 8 GPUs for testing and the others are still using `self.world_size` which is set to `TEST_GPU_NUM=4`. We need to unify this number later.

Code pointer:
https://github.com/pytorch/tau/blob/main/test/spmd/tensor/test_device_mesh.py#L20
https://github.com/pytorch/tau/blob/main/spmd/testing/common_utils.py#L31